pkg/resource/deploy/builtins.go

```diff
@@ -82,7 +85,10 @@ func (p *builtinProvider) Check(urn resource.URN, state, inputs resource.Propert
 		return nil, nil, fmt.Errorf("unrecognized resource type '%v'", urn.Type())
 	}
 
-	var name resource.PropertyValue
+	// We only need to warn about this in Check. This won't be called for Reads but Creates or Updates will call Check
+	// first.
+	p.diag.Warningf(diag.Message(urn, "The \"pulumi:pulumi:StackReference\" resource type is deprecated."))
```

Comment:
Can we tell users what to do in this case when they encounter this warning?

Reply:
Yeh it's an SDK bug so either update the SDK or if still seeing it on the latest SDK raise an issue.